kettle-plugins/common/ui/src/test/java/org/pentaho/big/data/plugins/common/ui/TestClusterTestDialog.java

```diff
@@ -0,0 +1,135 @@
+    RuntimeTestResultEntry entry = Mockito.mock( RuntimeTestResultEntry.class );
+
+    ArrayList<RuntimeTestModuleResults> results = new ArrayList<>();
+    results.add( runtimeTestModuleResults );
+    ArrayList<RuntimeTestResult> runtimeTestResults = new ArrayList<>();
+    runtimeTestResults.add( result );
+
+    Mockito.when( clusterTestStatus.getModuleResults() ).thenReturn( results );
+    Mockito.when( clusterTestStatus.isDone() ).thenReturn( true );
+    Mockito.when( runtimeTestModuleResults.getRuntimeTestResults() ).thenReturn( runtimeTestResults );
+    Mockito.when( result.getRuntimeTest() ).thenReturn( Mockito.mock( RuntimeTest.class ) );
+    Mockito.when( result.getRuntimeTestResultEntries() ).thenReturn( new ArrayList<>() );
+    Mockito.when( result.getOverallStatusEntry() ).thenReturn( entry );
+    Mockito.when( entry.getSeverity() ).thenReturn( RuntimeTestEntrySeverity.FATAL );
+    Mockito.when( entry.getException() ).thenReturn( new Exception() );
+
+    testDialog.getRunnable( progressBar, clusterTestStatus, testLabel ).run();
+
+    // check that exception is called two times: 1st - for null check, 2nd to print stack trace to log
+    Mockito.verify( entry, Mockito.times( 2 ) ).getException();
+  }
+
+  @Test
+  public void testExceptionsArePrintedToLog() throws KettleException {
+    ProgressBar progressBar = Mockito.mock( ProgressBar.class );
+    RuntimeTestStatus clusterTestStatus = Mockito.mock( RuntimeTestStatus.class );
+    Label testLabel = Mockito.mock( Label.class );
+    RuntimeTestModuleResults runtimeTestModuleResults = Mockito.mock( RuntimeTestModuleResults.class );
+    RuntimeTestResult result = Mockito.mock( RuntimeTestResult.class );
+    RuntimeTestResultEntry entry = Mockito.mock( RuntimeTestResultEntry.class );
+
+    ArrayList<RuntimeTestModuleResults> results = new ArrayList<>();
+    results.add( runtimeTestModuleResults );
+    ArrayList<RuntimeTestResult> runtimeTestResults = new ArrayList<>();
+    runtimeTestResults.add( result );
+    ArrayList<RuntimeTestResultEntry> entries = new ArrayList<>();
+    entries.add( entry );
+    entries.add( entry );
+
+    Mockito.when( clusterTestStatus.getModuleResults() ).thenReturn( results );
+    Mockito.when( clusterTestStatus.isDone() ).thenReturn( true );
+    Mockito.when( runtimeTestModuleResults.getRuntimeTestResults() ).thenReturn( runtimeTestResults );
+    Mockito.when( result.getRuntimeTest() ).thenReturn( Mockito.mock( RuntimeTest.class ) );
+    Mockito.when( result.getRuntimeTestResultEntries() ).thenReturn( entries );
+    Mockito.when( entry.getSeverity() ).thenReturn( RuntimeTestEntrySeverity.FATAL );
+    Mockito.when( entry.getException() ).thenReturn( new Exception() );
+
+    testDialog.getRunnable( progressBar, clusterTestStatus, testLabel ).run();
+
+    // check that exception is called four times: 2 - for null check, 2 to print stack trace to log
+    Mockito.verify( entry, Mockito.times( 4 ) ).getException();
```

Comment:
https://github.com/pentaho/pentaho-hadoop-shims/blob/master/common/hbase-comparators/src/test/java/org/pentaho/hbase/shim/common/CommonHBaseConnectionTest.java#L154
it looks better to mock LogChannel and check whether mock logBasic was called, not getException from entry